extension/inet/inet_escape_functions.cpp

```diff
@@ -0,0 +1,242 @@
+		if (sec_arg.validity.RowIsValid(0)) { // if not NULL
+			quote.Reinterpret(args.data[1]);
+		}
+	}
+	BinaryExecutor::Execute<string_t, bool, string_t>(args.data[0], quote, result, args.size(),
+	                                                  [&](string_t input_str, bool input_quote) {
+		                                                  auto str = input_str.GetData();
+		                                                  auto str_size = input_str.GetSize();
+		                                                  string escaped;
+		                                                  for (size_t i = 0; i < str_size; ++i) {
+			                                                  char ch = str[i];
+			                                                  switch (ch) {
+			                                                  case '&':
+				                                                  escaped += "&amp;";
+				                                                  break;
+			                                                  case '<':
+				                                                  escaped += "&lt;";
+				                                                  break;
+			                                                  case '>':
+				                                                  escaped += "&gt;";
+				                                                  break;
+			                                                  case '"':
+				                                                  if (input_quote) {
+					                                                  escaped += "&quot;";
+				                                                  } else {
+					                                                  escaped += "\"";
+				                                                  }
+				                                                  break;
+			                                                  case '\'':
+				                                                  if (input_quote) {
+					                                                  escaped += "&#x27;";
+				                                                  } else {
+					                                                  escaped += "\'";
+				                                                  }
+				                                                  break;
+			                                                  default:
+				                                                  escaped += ch;
+			                                                  }
+		                                                  }
+		                                                  return StringVector::AddString(result, escaped);
+	                                                  });
+}
+
+static void add_blob(int &code_point, string &result) {
+	int sz = 0;
+	char c[4] = {'\0', '\0', '\0', '\0'};
+	D_ASSERT(Utf8Proc::CodepointToUtf8(code_point, sz, c));
+	result += c;
+}
+
+static string ReplaceCharref(string &str) {
```

Comment:
Can we pass in a `string_t` instead? Note that constructing a `string` always involves copying the string data from the vector.